yapapi/network.py

```diff
@@ -54,6 +55,26 @@ def get_websocket_uri(self, port: int) -> str:
         return f"{net_api_ws}/net/{self.network.network_id}/tcp/{self.ip}/{port}"
 
 
+class NetworkStateMachine(StateMachine):
+    """State machine describing the states and lifecycle of a :class:`Network` instance."""
+
+    # states
+    initialized = State("initialized", initial=True)
+    creating = State("creating")
+    ready = State("ready")
+    removing = State("removing")
+    removed = State("removed")
+
+    # transitions
+    add_address = creating.to.itself() | ready.to.itself()
```

Comment:
hmm... I'm unsure I like how the state transitions are used to control when the methods can be run... I'd expect that logic to be part of those methods and not declared on the state machine ... is this some kind of a well-understood pattern? ... asking since I might not realize that this "transition" is used to control method's usage and rather think that it indeed somehow changes the state of the network...

Reply:
Not sure if this is an established pattern, I just found this more elegant.
Let's take `add_owner_address` as an example. If we wanted to ensure the correct state in the function directly this would look something like this:
```
        if self.state is not NetworkStateMachine.creating or self.state is not NetworkStateMachine.ready:
            raise SomeKindOfError("here")
```
I think that expressing this using possible transitions is cleaner and scales better if we need to handle different situations or new states in the future.
However, if you find this confusing I can think of a different approach.

Reply:
nah, maybe it's okay ...

Reply:
```suggestion
    add_owner_address = creating.to.itself() | ready.to.itself()
```

Reply:
changed in b319b73